### DISCUSSION TITLE
Help to implement an erosion coefficient in morphotectonics

### DISCUSSION THREAD

**Discussion by AllanLopez** [wschwanghart/topotoolbox#37] [Q&A]
Dear Wolfgang
Although erosion and the erosion coefficient are vital elements of the stream power incision model and its equation and it is also implicit in the Steepness, Chi and Hypsometric integral, there is a need for a relative reference value to correlate with the previous cited metrics. Meanwhile absolute values determined from cosmogenic nucleides are expensive. 
The authors of the following paper proposed an erosion coefficient K= E/ Ksn, directly derived from TTT, but I unable to implement it due to my Matlab deep limitations, so I ask you please Wolfgang, to do it along with a Shp export capability. 
My deep appreciation in advance
Allan
Impact of fault damage on eastern Tibet topography Heather M. Kirkpatrick, Seulgi Moon*, An Yin and T. Mark Harrison (Geology, 2020; open access at https://pubs.geoscienceworld.org/gsa/geology/article/49/1/30/590535/Impact-of-fault-damage-on-eastern-Tibet-topography)
They mention that:
You can extract channel steepness (ksn) from "chiplot" function in topo toolbox.

Then, you can use fitlm or fitnlm to fit ksn and erosion rate and examine which fit explain the overall relationship.
Once you decide your functional form, you can calculate erosion coefficient using the best-fit parameters (e.g., power-law exponents)

for example, 
xda1 = ksn; yda1 = erosion;
beta0 = [1];
nonlm0 = fitnlm(xda1, yda1 , @(b,xda1)b(1).*xda1.^1,beta0); %linear without intercept

beta0 = [1e-2 1];
nonlm1 = fitnlm(xda1, yda1 , @(b,xda1)exp(xda1.*b(1) +b(2)),beta0) ;%exponential
 
beta0 = [1e-2 1];
nonlm2 =fitnlm(xda1, yda1 , @(b,xda1)b(1).*xda1.^b(2),beta0) ; %power law

beta0 = [1 0];
nonlm3 = fitnlm(xda1, yda1 , @(b,xda1)b(1).*xda1.^1+b(2),beta0); %linear with intercept

b_n2 = nonlm2.Coefficients{:,1}
erocoef_n2 = erosion./(ksn.^b_n2(2)); %n = 0.49, coefficient for nonlinear powerlaw fit 
erocoef = erosion./(ksn); %n = 0.49, coefficient for linear fit without intercept 

Hope this helps!
Seulgi

**Comment by AllanLopez** [comment]
[Impact of fault damage on eastern Tibet topography.pdf](https://github.com/wschwanghart/topotoolbox/files/11646643/Impact.of.fault.damage.on.eastern.Tibet.topography.pdf)

**Comment by wschwanghart** [comment]
Dear Allan,
you have been following my blog for years, and you are still no MATLAB expert? What have I done wrong? ;-) (just joking)
Please give me some sample code, so that I can help. What have you done so far to solve the problem. 
Cheers, Wolfgang

**Comment by AllanLopez** [comment]
Dear Wolfgang,
Thank you for your response. After all these years I have learned to detect
my mistakes in the execution, I hope.
What is obvious for you is martian language to me (also joking).

I guess a bridge is needed to fill the gap between Ksn extracted from
chipplot and the following example given by the author below. I mean I have
not worked anything; I do not know what to do.

Later I shall mention and idea to visit you for a couple of weeks next year.

All the best,

Allan

for example,

Then, you can use fitlm or fitnlm to fit ksn and erosion rate and examine
which fit explain the overall relationship.

Once you decide your functional form, you can calculate erosion coefficient
using the best-fit parameters (e.g., power-law exponents)


xda1 = ksn; yda1 = erosion;

beta0 = [1];

nonlm0 = fitnlm(xda1, yda1 , @(b,xda1)b(1).*xda1.^1,beta0); %linear without
intercept



beta0 = [1e-2 1];

nonlm1 = fitnlm(xda1, yda1 , @(b,xda1)*exp*(xda1.*b(1) +b(2)),beta0)
;%exponential



beta0 = [1e-2 1];

nonlm2 =fitnlm(xda1, yda1 , @(b,xda1)b(1).*xda1.^b(2),beta0) ; %power law



beta0 = [1 0];

nonlm3 = fitnlm(xda1, yda1 , @(b,xda1)b(1).*xda1.^1+b(2),beta0); %linear
with intercept



b_n2 = nonlm2.Coefficients{:,1}

erocoef_n2 = erosion./(ksn.^b_n2(2)); %n = 0.49, coefficient for nonlinear
powerlaw fit

erocoef = erosion./(ksn); %n = 0.49, coefficient for linear fit without
intercept
